content/en/docs/concepts/storage/volumes.md

```diff
@@ -1160,6 +1160,33 @@ spec:
 
 More examples can be found [here](https://github.com/kubernetes/examples/tree/master/staging/volumes/vsphere).
 
+#### CSI Migration
+
+{{< feature-state for_k8s_version="v1.19" state="beta" >}}
+
+The CSI Migration feature for vsphereVolume, when enabled, shims all plugin operations
+from the existing in-tree plugin to the `csi.vsphere.vmware.com` {{< glossary_tooltip text="CSI" term_id="csi" >}} Driver. In order to use this feature, the [vSphere CSI
+Driver](https://github.com/kubernetes-sigs/vsphere-csi-driver)
+must be installed on the cluster and the `CSIMigration` and `CSIMigrationvSphere`
+features must be enabled. This also requires minimum vSphere vCenter/ESXi Version to be 7.0u1 and minimum HW Version to be VM version 15.
+
+The following StorageClass parameters from vsphereVolume plugin will not be supported by vSphere CSI driver:
+
+* diskformat
+* hostfailurestotolerate
+* forceprovisioning
+* cachereservation
+* diskstripes
+* objectspacereservation
+* iopslimit
+
+Existing volumes created using these parameters will be migrated to the vSphere CSI driver, but new volumes created by the vSphere CSI driver will not be honoring these parameters.
+
+#### CSI Migration Complete
+{{< feature-state for_k8s_version="v1.19" state="beta" >}}
+
+To turn off the vsphereVolume plugin from being loaded by controller manager and kubelet, you need to set this feature flag to true. This requires `csi.vsphere.vmware.com` {{< glossary_tooltip text="CSI" term_id="csi" >}} driver being installed on all worker nodes.
+
```

Comment:
What does this mean? I wasn't sure, especially the heading.
Bear in mind that this is conceptual documentation - would it make sense to add a [blog](https://kubernetes.io/blog/) post about how to migrate, rather than include this section here?

Reply:
This is a separate feature gate that will disable the loading of the legacy driver entirely. The difference is subtle but there is some fallback logic in the implementation that will handle the scenario where the nodes are still running the legacy driver. Turning on the migration complete feature gate disables that fallback logic.

Reply:
Also this section is mostly copied form the wording that we use for the other csi migration features. If you think the wording is confusing, we should work on updating the wording for all the cloud providers.

Reply:
OK, I've logged https://github.com/kubernetes/website/issues/22521 and am happy given that to skip fixing it here.